spinalcordtoolbox/moco.py

```diff
@@ -865,3 +868,209 @@ def spline(folder_mat, nt, nz, verbose, index_b0 = [], graph=0):
+
+    # reorient back to original orientation
+    if im_data_merged.orientation != orig_orientation:
+        im_data_merged.change_orientation(orientation=orig_orientation, generate_path=True)
+    # Save to tmp dir
+    im_data_merged.save(sct.add_suffix(file_data, '_merged'), verbose=0)
+
+    # Generate b0_moco_mean and dwi_moco_mean
+    args = ['-i', os.path.join(path_tmp,sct.add_suffix(file_data, '_merged')),
+            '-bvec', param.fname_bvecs, '-a', '1', '-v', '0']
+    if not param.fname_bvals == '':
+        # if bvals file is provided
+        args += ['-bval', param.fname_bvals]
+    fname_b0, fname_b0_mean, fname_dwi, fname_dwi_mean = sct_dmri_separate_b0_and_dwi.main(args=args)
+
+    # Deal with motion parameters
+    if param.output_motion_param:
+        # nii motion parameters
+        moco_params_x_odd = Image('moco_params_x_odd.nii.gz')
+        moco_params_x_even = Image('moco_params_x_even.nii.gz')
+        moco_params_x_merged = np.zeros((1, 1, nz, nt), dtype=np.float32)     # create empty array for x motion params
+        moco_params_y_odd = Image('moco_params_y_odd.nii.gz')
+        moco_params_y_even = Image('moco_params_y_even.nii.gz')
+        moco_params_y_merged = np.zeros((1, 1, nz, nt), dtype=np.float32)     # create empty array for y motion params
+        counter_even = 0
+        counter_odd = 0
+        for index in range(0, im_data_merged.dim[2]):
+            if index % 2 == 0:
+                moco_params_x_merged[:, :, index, :] = moco_params_x_even.data[:, :, counter_even, :]
+                moco_params_y_merged[:, :, index, :] = moco_params_y_even.data[:, :, counter_even, :]
+                counter_even += 1
+            elif index % 2 != 0:
+                moco_params_x_merged[:, :, index, :] = moco_params_x_odd.data[:, :, counter_odd, :]
+                moco_params_y_merged[:, :, index, :] = moco_params_y_odd.data[:, :, counter_odd, :]
+                counter_odd += 1
+        # Save nii to tmp dir
+        import nibabel as nib
+        # TODO - merged moco params files have isotropic 1mm voxel size instead of voxel size of original image
+        img_params_x = nib.Nifti1Image(moco_params_x_merged, None)
+        nib.save(img_params_x, sct.add_suffix(file_moco_params_x, '_merged'))
+        img_params_y = nib.Nifti1Image(moco_params_y_merged, None)
+        nib.save(img_params_y, sct.add_suffix(file_moco_params_y, '_merged'))
+
+        # tsv motion parameters (slice-wise average)
+        moco_param = np.concatenate((np.squeeze(np.mean(moco_params_x_merged, 2)).reshape(-1, 1),
+                                     np.squeeze(np.mean(moco_params_y_merged, 2)).reshape(-1, 1)), axis=1)
+        with open(sct.add_suffix(file_moco_params_csv, '_merged'), 'wt') as out_file:
+            tsv_writer = csv.writer(out_file, delimiter='\t')
+            tsv_writer.writerow(['X', 'Y'])
+            for mocop in moco_param:
+                tsv_writer.writerow([mocop[0], mocop[1]])
+
```

Comment:
you should add a unit test for this

Reply:
Okay. It means to extend existing [test_sct_dmri_moco.py](https://github.com/neuropoly/spinalcordtoolbox/blob/master/testing/test_sct_dmri_moco.py) script by interleaved data, right?